src/containers/TimerContainer/TimerContainer.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react'
+
+import Timer from '../../components/Timer/Timer'
+
+export default class TimerContainer extends Component {
+  constructor(props) {
+    super(props)
+    this.nextDate = new Date('2017-03-13 22:10:00')
+    this.state = {
+      timeLeft: (this.nextDate - Date.now() > 0) ? this.nextDate - Date.now() : 0
+    }
+  }
+
+  stringifyTime(t) {
+    return (t < 10) ? '0' + t : t
+  }
+
+  componentDidMount() {
+    this.timer = (this.state.timeLeft > 0) ? setInterval(this.tick, 1000) : null
+  }
+
+  componentWillUpdate() {
+    if (this.state.timeLeft <= 1000) {
+      clearInterval(this.timer)
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer)
+  }
+
+  tick() {
+    this.setState({ timeLeft: Math.floor((this.nextDate - Date.now()) / 1000) * 1000 })
+  }
```

Comment:
How does this work without binding?

Reply:
Not sure, but it does! Changing to an arrow function anyway